builders/build/eth-api/verify-contracts/api-verification.md

```diff
@@ -0,0 +1,530 @@
+---
+title: Verify Smart Contracts through APIs
+description: Learn how to verify smart contracts on Moonbeam-based networks using one of the available API-based verification methods. 
+---
+
+# API-based Contract Verification
+
+![Explorer Banner](/images/builders/build/eth-api/verify-contracts/block-explorers/verify-contract-banner.png)
+
+## Introduction {: #introduction } 
+
+Verifying smart contracts greatly improves their transparency and security.  Smart contracts deployed on Moonbeam networks can be verified through API-based tools, including Moonscan API and Sourcify. 
+
+This page will outline the steps for using these API-based tools for verifying smart contracts, or retrieving verification status and metadata of smart contracts on Moonbeam networks.
+
+## Using Moonscan API {: #using-moonscan-api } 
+
+[Moonscan](https://moonscan.io/){target=_blank} is an official fork of Etherscan that can be used to view and search on-chain data, and comes with a suite of developer tools and analytics to interact with data on Moonbeam networks. 
+
+Moonscan exposes a set of REST API endpoints similar to the [Etherscan API](https://docs.etherscan.io/){target=_blank} endpoints that can be used to verify smart contracts, retrieve verified contract ABI and source code, or otherwise interact with verified contracts on Moonbeam networks. 
+
+### Generating a Moonscan API Key {: #generating-a-moonscan-api-key }
+
+Before using the Moonscan API, you need to generate a Moonscan API key. Please follow the instructions in [the key generation section](builders/build/eth-api/verify-contracts/etherscan-plugins/#generating-a-moonscan-api-key){target=_blank} of the Etherscan plug-in verification page, as the API keys generated are used for both.
+
+### Moonscan Public API URL {: #moonscan-public-api-url }
+
+The Moonscan API URL for Moonbeam networks is as follows:
+
+=== "Moonbeam"
+    ```
+    https://api-moonbeam.moonscan.io/api
+    ```
+
+=== "Moonriver"
+    ```
+    https://api-moonriver.moonscan.io/api
+    ```
+
+=== "Moonbase Alpha"
+    ```
+    https://api-moonbase.moonscan.io/api
+    ```
+
+### Verify Source Code {: #verify-source-code }
+
+To verify a deployed contract's source code using the Moonscan API, you must form a POST request containing all the relevant contract creation information, and send the request to Moonscan's REST API. The following is sample code using Javascript:
+
+=== "Moonbeam"
+    ```javascript
+    //Submit Source Code for Verification
+    $.ajax({
```

Comment:
is this jquery?

Reply:
there should be a space after all of these `//` (for example: `// Only POST supported` and all comments should start with capitalized letter

Reply:
> is this jquery?

Maybe, I'm not too clear on the difference between jquery javascript and AJAX lol. The javascript linter(?) seems to pick up the syntax.